### DISCUSSION TITLE
Pydantic v2.3 equivalent of overriding field names in subclasses using old style Config class 'fields' setting.

### DISCUSSION THREAD

**Discussion by wyattcolyn** [pydantic/pydantic#7248] [Question]
In previous versions of pydantic (less than 2.0?) it was possible to override the field name of an inherited model using the 'fields' setting in the Config class. Example:

```python
class CoolParentClass(BaseModel):
    coolfield: str = Field(..., description='a cool field', example='hello!')

class CoolSubClass(CoolParentClass):
    class Config:

        fields = {'coolfield': 'the_coolfield'}
``` 

Creating a CoolSubClass like so 
```python
CoolSubClass(the_coolfield='hello')
``` 
results in a successful validation. Note that this is different than using an alias. 

This tactic easily allows hierarchical structures to be built with several top level parent models. I am working on upgrading pydantic in a large codebase that uses this pattern everywhere, and so I'd like to keep this behavior if possible. 

According to the migration guide the 'fields' setting was removed, but it suggests the same effect can be achieved using Annotated: https://docs.pydantic.dev/latest/migration/#changes-to-config

I can't seem to find anything that suggests how to use Annotated to get this same effect. Is there some other way without Annotated? If anyone has figured out a way, please share your findings!

**Comment by mohammadali95** [comment]
Is this possible in V2? Trying to achieve the same behavior as well! Any suggestions @sydney-runkle ?

**Comment by sydney-runkle** [comment]
@mohammadali95,

Good question. There's not currently a smooth workaround. Feel free to open an issue with a feature request - this is something that I think a lot of folks might appreciate having reintroduced. We'd just have to do some digging to figure out how type safe we could make the changes...
